x-pack/plugin/core/src/main/java/org/elasticsearch/license/PostStartTrialResponse.java

```diff
@@ -50,32 +66,148 @@ RestStatus getRestStatus() {
             return restStatus;
         }
 
+        @Override
+        public String toString() {
+            return this.name().toLowerCase(Locale.ROOT);
+        }
+
+        public static Status fromString(String value) {
+            return valueOf(value.toUpperCase(Locale.ROOT));
+        }
+
+    }
+
+    private static final ParseField TRIAL_WAS_STARTED_FIELD = new ParseField("trial_was_started");
+    private static final ParseField TRIAL_STATUS_FIELD = new ParseField("trial_status");
+    private static final ParseField LICENSE_TYPE_FIELD = new ParseField("type");
+    private static final ParseField ERROR_MESSAGE_FIELD = new ParseField("error_message");
+    private static final ParseField ACKNOWLEDGE_DETAILS_FIELD = new ParseField("acknowledge");
+    private static final ParseField ACKNOWLEDGE_HEADER_FIELD = new ParseField("message");
+
+    private static final ConstructingObjectParser<PostStartTrialResponse, Void> PARSER = new ConstructingObjectParser<>(
+      "post_start_trial_response", true, (Object[] arguments, Void aVoid) -> {
+
+          // the acknowledge, trial_was_started, and error_message fields are inferred from the trial status
+          final Status trialStatus = Status.fromString((String) arguments[0]);
+          final String licenseType = (String) arguments[1];
+
+          @SuppressWarnings("unchecked")
+          final Tuple<String, Map<String, String[]>> acknowledgeDetails = (Tuple<String, Map<String, String[]>>) arguments[2];
+
+          if (acknowledgeDetails == null) {
+              return new PostStartTrialResponse(trialStatus, licenseType);
+          } else {
+              return new PostStartTrialResponse(trialStatus, licenseType, acknowledgeDetails.v2(), acknowledgeDetails.v1());
+          }
+    });
+
+    static {
+        PARSER.declareString(constructorArg(), TRIAL_STATUS_FIELD);
+        PARSER.declareString(constructorArg(), LICENSE_TYPE_FIELD);
+        // todo when PutLicenseResponse returns to this project, consolidate this acknowledgement parsing code with it
+        PARSER.declareObject(
+            optionalConstructorArg(),
+            (parser, v) -> {
+                Map<String, String[]> acknowledgeMessages = new HashMap<>();
```

Comment:
It should be, but in some cases it might not be possible. For instance if fields content differ widely depending on the data in the XContent itself, sometimes it is easier to go back to old school parsing. That being said, I need to look at this to see what it is doing to see if *this instance* is needed